src/envoy/utils/authn.h

```diff
@@ -40,6 +41,11 @@ class Authentication : public Logger::Loggable<Logger::Id::filter> {
   static bool FetchResultFromHeader(const Http::HeaderMap& headers,
                                     istio::authn::Result* result);
 
+  // Returns a pointer to the authentication result from request info, if
+  // available. Otherwise, return nullptrl
+  static const ProtobufWkt::Struct* GetResultFromRequestInfo(
```

Comment:
can we not use Struct? just return the exact proto message?

Reply:
The authn result stored in request info is already converted into Struct (key-value pairs of attributes). It was implemented by @yangminzhu and use for RBAC. We will change mixer code to consume this struct and populate attributes from this.